singularity-opac/photons/gray_opacity_photons.hpp

```diff
@@ -0,0 +1,81 @@
+// ======================================================================
+// © 2021. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+// ======================================================================
+
+#ifndef SINGULARITY_OPAC_PHOTONS_GRAY_OPACITY_PHOTONS_
+#define SINGULARITY_OPAC_PHOTONS_GRAY_OPACITY_PHOTONS_
+
+#include <cassert>
+#include <cmath>
+#include <cstdio>
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-opac/base/opac_error.hpp>
+#include <singularity-opac/photons/thermal_distributions_photons.hpp>
+
+namespace singularity {
+namespace photons {
+
+class GrayOpacity {
+ public:
+  GrayOpacity(const Real kappa) : kappa_(kappa) {}
+  GrayOpacity(const PlanckDistribution &dist, const Real kappa)
+      : dist_(dist), kappa_(kappa) {}
+
+  GrayOpacity GetOnDevice() { return *this; }
+  PORTABLE_INLINE_FUNCTION
+  int nlambda() const noexcept { return 0; }
+  PORTABLE_INLINE_FUNCTION
+  void PrintParams() const noexcept { printf("Gray opacity. kappa = %g\n", kappa_); }
+  inline void Finalize() noexcept {}
+
+  PORTABLE_INLINE_FUNCTION
+  Real OpacityPerNu(const Real rho, const Real temp, const Real nu,
+                    Real *lambda = nullptr) const {
+    return dist_.OpacityFromKirkhoff(*this, rho, temp, nu, lambda);
+  }
+
+  PORTABLE_INLINE_FUNCTION
+  Real EmissivityPerNuOmega(const Real rho, const Real temp, const Real nu,
+                            Real *lambda = nullptr) const {
+    Real Bnu = dist_.ThermalDistributionOfTNu(temp, nu, lambda);
+    return rho * kappa_ * Bnu;
+  }
+
+  PORTABLE_INLINE_FUNCTION
+  Real EmissivityPerNu(const Real rho, const Real temp, const Real nu,
+                       Real *lambda = nullptr) const {
+    return 4 * M_PI * EmissivityPerNuOmega(rho, temp, nu, lambda);
+  }
+
+  PORTABLE_INLINE_FUNCTION
+  Real Emissivity(const Real rho, const Real temp, Real *lambda = nullptr) const {
+    Real B = dist_.ThermalDistributionOfT(temp, lambda);
+    return rho * kappa_ * B;
+  }
+
+  PORTABLE_INLINE_FUNCTION
+  Real NumberEmissivity(const Real rho, const Real temp, Real *lambda = nullptr) const {
+    return kappa_ * dist_.ThermalNumberDistribution(temp, lambda);
+  }
+
+ private:
+  Real kappa_; // absorption coefficient. Units of 1/cm
```

Comment:
Should we just call this an opacity with units cm^2/g? Looks like `rho` is always available to make that conversion.

Reply:
Yes. This should be an opacity. I'm multiplying everything by rho everywhere. I'll change the comment.